modules/gmake2/tests/test_gmake2_clangtidy.lua

```diff
@@ -0,0 +1,45 @@
+--
+-- test_gmake2_clangtidy.lua
+-- Test ClangTidy support in Makefiles.
+--
+
+local suite = test.declare("gmake2_clangtidy")
+local gmake2 = premake.modules.gmake2
+
+local wks, prj, cfg
+
+function suite.setup()
+	wks = workspace("MyWorkspace")
+	configurations { "Debug", "Release" }
+  targetname "blink"
+	kind "StaticLib"
+	language "C++"
+	prj = test.createProject(wks)
+end
+
+local function prepare()
+	wks = test.getWorkspace(wks)
+	prj = test.getproject(wks, 1)
+	cfg = test.getconfig(prj, "Debug")
+  gmake2.cpp.clangtidy(cfg, toolset)
+  files { "src/hello.cpp", "src/hello2.c" }
+
+end
+
+function suite.clangtidyOn()
+  clangtidy "On"
+
+	prepare()
+
+  test.capture [[
+CLANG_TIDY = clang-tidy
+  ]]
+end
+
+function suite.clangtidyOff()
```

Comment:
```suggestion
function suite.clangtidyDefaulted()
```
currently, and you might add a dedicated test for `"Off"`, especially when code suggest that it is wrongly handled ;-)